Update UI screenshots in Pipeline documentation #7526

ShrushtiDesai opened the pull request:
This PR addresses issue #7526.

I have updated the screenshots for the pages under the Pipeline section of the documentation. Additionally, I have revised the alt text for the screenshots and made necessary adjustments to the associated text in the documentation where applicable.

kmartens27 commented:
Hi @ShrushtiDesai thank you again for submitting this. I'll be reviewing it today, as soon as I have time to dedicate to it. I'll be sure to check and provide any feedback that I might come up with.

kmartens27 commented on content/doc/book/pipeline/shared-libraries.adoc:
```diff
@@ -94,7 +94,7 @@ There are several places where Shared Libraries can be defined, depending on
 the use-case. _Manage Jenkins » System » Global Pipeline Libraries_
 as many libraries as necessary can be configured.
```
```suggestion
the use-case. 
Navigate to _Manage Jenkins » System » Global Trusted Pipeline Libraries_ to configure as many libraries as necessary.
```
Suggestion for text update for grammar and to match the new UI

kmartens27 commented on content/doc/book/pipeline/pipeline-as-code.adoc:
```diff
@@ -203,37 +203,36 @@ To demonstrate using an Organization Folder to manage repositories, we'll use
-Make sure the *owner* matches the GitHub Organization name exactly, in our case it must be: _cloudbeers_.
+Make sure the *owner* matches the GitHub Organization name exactly, in our case it must be: _CloudBeersInc_.
 This defaults to the same value that was entered for the item name in the first step.
 Next, select or add new "Scan credentials" - we'll enter our GitHub username and access token as the password.
```
```suggestion
Next, select or add new *Credentials* - we'll enter our GitHub username and access token as the password.
```
Adjustment to match new UI

kmartens27 reviewed (changes requested):
Thank you very much @ShrushtiDesai for all of this work! A lot of this is great and definitely improves upon what the documentation currently has. Most of the screenshots look good and appear to be compressed/reducing the size of the page, so thank you very much for all of that.

I've provided a few examples of where the text needs to change to match the updated UI.

I've also found a couple screenshots that need some modifcation:

- [content/doc/book/resources/pipeline/configure-global-pipeline-library.png] and [content/doc/book/resources/pipeline/global-pipeline-library-modern-scm.png] are using the term "master". While that is accurate, it would be ideal if these could be replaced with "main" to align with the [inclusive naming initiative](https://www.jenkins.io/sigs/advocacy-and-outreach/#inclusive-naming). It's not entirely possible to just rename this within Jenkins core, as it could break or negatively affect things already built upon it. However, since the documentation is using examples and for informational purposes, using main would follow the project's direction.

- [content/doc/book/resources/pipeline/snippet-generator.png] would it be possible to use an arrow shape tool instead of drawing it manually? The pixelation on the arrow does not look as clean as it could.

Please let me know if you have any questions or concerns about this!

ShrushtiDesai commented:
Hi @kmartens27,

I’ve committed the text updates based on your suggestions and updated the screenshots to align with the naming convention, replacing "master" with "main" where applicable.

I also noticed that on the "Pipeline as Code" page, the Scan Organization process for CloudBeersInc displays three repositories, two of which still reference the "master" branch, as well as the PR demo image where the "master" branch is shown. I’ve updated those as well, along with the alt text for accessibility.

I see that the first check in the pull request has failed. If there are any specific modifications needed to resolve this, please let me know, and I’ll make the necessary adjustments.

Additionally, a merge conflict has arisen since I force-pushed the branch updates. I’m not entirely sure how to resolve this conflict—could you provide some guidance?

kmartens27 commented:
Hi @ShrushtiDesai, I see the commit where this all happened and I've done this myself before so I know it can be a bit tricky.

The instructions shared in [this community discussion](https://github.com/orgs/community/discussions/64693) explain how you can use `git reflog` to identify the commit/state before the force push, and then utilizing the `git reset` action along with the commit id, you would be able to reset to before the force push. Generally, it is best to avoid force pushing unless you are certain that everything else is up to date.

There are other instructions and guides, such as [this medium article](https://waresix.engineering/undoing-a-force-push-in-git-recovering-your-git-repository-safely-f24dcd98cf0f), but they all generally advise the same steps. Alternatively, [this guide](https://www.jvt.me/posts/2021/10/23/undo-force-push/) shows how to find the information from the GitHub UI.

I have also saved my work somewhere outside of the repository (in a text editor or other IDE software) and created a new branch to import the work to. This would not be the worst option, but would require you to re-do the work which may be more involved than the reset.

If I can come up with any other methods I will be sure to share here, but these would be your best options at this time.

ShrushtiDesai commented:
Hi @kmartens27,

It seems I made a mistake when updating my branch, which caused it to merge with master, bringing in other commits that weren’t intended for this branch. I apologize for the inconvenience this may have caused. I’ll redo these updates from a clean branch. I think the best option would be to close this pull request and open a new one.

kmartens27 commented:
@ShrushtiDesai whatever makes the most sense and would be easiest to accomplish is more than alright. 

I try to make sure that I've alway run through `git fetch --all`, 'git pull --all', and 'gh repo sync'  on the master/main branch before I do any work so that everything is as up to date as possible prior to me starting. If other commits are made outside of your branch and never pulled in, this sort of thing can happen when the branches are updated.
